src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -0,0 +1,101 @@
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var eventHubsNamespace = new EventHubsNamespace(construct, name: name);
+
+            eventHubsNamespace.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
+
+            eventHubsNamespace.AssignProperty(p => p.Sku.Name, new Parameter("sku", defaultValue: "Standard"));
+
+            var eventHubsDataOwnerRole = eventHubsNamespace.AssignRole(RoleDefinition.EventHubsDataOwner);
+            eventHubsDataOwnerRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
+
+            eventHubsNamespace.AddOutput("eventHubsEndpoint", sa => sa.ServiceBusEndpoint);
+
+            var azureResource = (AzureEventHubsResource)construct.Resource;
+            var azureResourceBuilder = builder.CreateResourceBuilder(azureResource);
+            configureResource?.Invoke(azureResourceBuilder, construct, eventHubsNamespace);
+
+            foreach (var hub in azureResource.Hubs)
+            {
+                var hubResource = new EventHub(construct, name: hub.Name, parent: eventHubsNamespace);
+                hub.Configure?.Invoke(azureResourceBuilder, construct, hubResource);
+            }
+            
+        };
+        var resource = new AzureEventHubsResource(name, configureConstruct);
+
+        return builder.AddResource(resource)
+                      // These ambient parameters are only available in development time.
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
+                      .WithManifestPublishingCallback(resource.WriteToManifest);
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs hub resource to the application model. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="name">The name of the Event Hub.</param>
+    public static IResourceBuilder<AzureEventHubsResource> AddEventHub(this IResourceBuilder<AzureEventHubsResource> builder, string name)
+    {
+#pragma warning disable CA2252 // This API requires opting into preview features
+        return builder.AddEventHub(name, null);
+#pragma warning restore CA2252 // This API requires opting into preview features
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs hub resource to the application model. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="name">The name of the Event Hub.</param>
+    /// <param name="configureHub">Optional callback to customize the Event Hub.</param>
+    [RequiresPreviewFeatures]
```

Comment:
These should be `[Experimental]` instead. See https://github.com/dotnet/aspire/issues/2918